packages/web3-providers-ws/test/integration/web_socket_provider_integration.test.ts

```diff
@@ -0,0 +1,338 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* eslint-disable jest/no-done-callback */
+import {
+	EthExecutionAPI,
+	JsonRpcId,
+	Web3APIPayload,
+	DeferredPromise,
+	JsonRpcResponse,
+	JsonRpcNotification,
+	JsonRpcSubscriptionResult,
+} from 'web3-common';
+import express from 'express';
+import { createProxyMiddleware } from 'http-proxy-middleware';
+
+import { Server } from 'http';
+import { Web3WSProviderError } from 'web3-errors';
+import WebSocketProvider from '../../src/index';
+// import { Web3WSProviderError } from '../../src/errors';
+import { WSRequestItem, OnCloseEvent } from '../../src/types';
+
+import {
+	getSystemTestProvider,
+	describeIf,
+	getSystemTestAccounts,
+} from '../fixtures/system_test_utils';
+
+describeIf(getSystemTestProvider().includes('ws'))(
+	'WebSocketProvider - implemented methods',
+	() => {
+		let clientWsUrl: string;
+		let accounts: string[];
+		let webSocketProvider: WebSocketProvider;
+		let jsonRpcPayload: Web3APIPayload<EthExecutionAPI, 'eth_getBalance'>;
+		// helper function
+		let currentAttempt = 0;
+		const waitForOpenConnection = async (provider: WebSocketProvider, status = 'connected') => {
+			return new Promise<void>((resolve, reject) => {
+				const maxNumberOfAttempts = 10;
+				const intervalTime = 5000; // ms
+
+				const interval = setInterval(() => {
+					if (currentAttempt > maxNumberOfAttempts - 1) {
+						clearInterval(interval);
+						reject(new Error('Maximum number of attempts exceeded'));
+					} else if (provider.getStatus() === status) {
+						clearInterval(interval);
+						resolve();
+					}
+					// eslint-disable-next-line no-plusplus
+					currentAttempt++;
+				}, intervalTime);
+			});
+		};
+		beforeAll(async () => {
+			clientWsUrl = getSystemTestProvider();
+			accounts = await getSystemTestAccounts();
+		});
+		describe('tests', () => {
```

Comment:
I don't think this `describe` is necessary

Reply:
I don't think you really need the other `describe`s as well (`describe('websocker provider tests'` and `describe('subscribe event tests'`) I suppose it doesn't matter that much if there is no performance hit though

Reply:
Some of them can be removed, but others are needed because I need different initialization in `beforeAll/Each`. Maybe I could separate them completely.

Reply:
Make some changes, now there is only one